allennlp/models/encoder_decoders/nlvr/nlvr_semantic_parser.py

```diff
@@ -73,43 +68,45 @@ class NlvrSemanticParser(Model):
         starting at the apropriate epoch.  The weight will remain constant if this is not provided.
     penalize_non_agenda_actions : ``bool``, optional (default=False)
         Should we penalize the model for producing terminal actions that are outside the agenda?
+    rule_namespace : ``str``, optional (default=rule_labels)
+        The vocabulary namespace to use for production rules.  The default corresponds to the
+        default used in the dataset reader, so you likely don't need to modify this.
     """
     def __init__(self,
                  vocab: Vocabulary,
                  sentence_embedder: TextFieldEmbedder,
-                 nonterminal_embedder: TextFieldEmbedder,
-                 terminal_embedder: TextFieldEmbedder,
+                 action_embedding_dim: int,
                  encoder: Seq2SeqEncoder,
                  attention_function: SimilarityFunction,
                  beam_size: int,
                  max_decoding_steps: int,
                  normalize_beam_score_by_length: bool = False,
                  checklist_cost_weight: float = 0.8,
                  dynamic_cost_weight: Dict[str, Union[int, float]] = None,
-                 penalize_non_agenda_actions: bool = False) -> None:
+                 penalize_non_agenda_actions: bool = False,
+                 rule_namespace: str = 'rule_labels') -> None:
         super(NlvrSemanticParser, self).__init__(vocab=vocab)
 
         self._sentence_embedder = sentence_embedder
         self._denotation_accuracy = Average()
         self._consistency = Average()
         self._agenda_coverage = Average()
-        check_dimensions_match(nonterminal_embedder.get_output_dim(),
-                               terminal_embedder.get_output_dim(),
-                               "nonterminal embedding dim",
-                               "terminal embedding dim")
-        self._nonterminal_embedder = nonterminal_embedder
-        self._terminal_embedder = terminal_embedder
         self._encoder = encoder
         self._decoder_trainer: DecoderTrainer[Callable[[NlvrDecoderState], torch.Tensor]] = \
                 ExpectedRiskMinimization(beam_size, normalize_beam_score_by_length, max_decoding_steps)
-        action_embedding_dim = nonterminal_embedder.get_output_dim() * 2
+        self._rule_namespace = rule_namespace
+
+        self._action_embedder = Embedding(num_embeddings=vocab.get_vocab_size(self._rule_namespace),
+                                          embedding_dim=action_embedding_dim)
+        self._initial_action_embedding = torch.nn.Parameter(torch.FloatTensor(action_embedding_dim))
```

Comment:
A comment about this field would be nice here. Also, I think `start_action_embedding` may be a better name, since this sounds like this is the initial state of the action embedding weights.

Reply:
`first_action_embedding`?  Just so there's no confusion with `START -> c`, `START -> d`, etc.